cmd/satellite/main.go

```diff
@@ -67,20 +109,79 @@ type satellite struct {
 	gsAddr   string
 }
 
+type flags struct {
+	fakeGPUs bool
+}
+
+func saveState(filename string, items []uplink.GPUStats) error {
+	data, err := json.Marshal(items)
+
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(filename, data, 0644)
+}
+
+func recoverState(filename string) ([]uplink.GPUStats, error) {
+	data, err := os.ReadFile(filename)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var state []uplink.GPUStats
+
+	err = json.Unmarshal(data, &state)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return state, nil
+}
+
+// Dummy function to process list of stats
+func processStats(stats []uplink.GPUStats) uplink.GPUStats {
+	return stats[len(stats)-1]
+}
+
+func parseProgramFlags() flags {
+	fakeGpus := flag.Bool("fakegpu", false, "Use fake GPU data")
+	flag.Parse()
+
+	return flags{
+		fakeGPUs: *fakeGpus,
+	}
+}
+
+func setGPUHandler(isFakeGPUs bool) gpustats.GPUDataSource {
```

Comment:
Why's this named this? `setGPUHandler` makes it sound like it's a setter for some state, when it isn't. How about `makeGpuHandler`?

Reply:
I refactored this out as it was setting state rather than something resembling a factory method. I think set might be more appropriate here.